src/Sylius/Behat/Element/Shop/Account/RegisterElement.php

```diff
@@ -51,43 +51,45 @@ public function getEmail(): string
 
     public function specifyFirstName(?string $firstName): void
     {
-        $this->getDocument()->fillField('First name', $firstName);
+        $this->getElement('first_name')->setValue($firstName);
     }
 
     public function specifyLastName(?string $lastName): void
     {
-        $this->getDocument()->fillField('Last name', $lastName);
+        $this->getElement('last_name')->setValue($lastName);
     }
 
     public function specifyPassword(?string $password): void
     {
-        $this->getDocument()->fillField('Password', $password);
+        $this->getElement('password')->setValue($password);
     }
 
     public function specifyPhoneNumber(string $phoneNumber): void
     {
-        $this->getDocument()->fillField('Phone number', $phoneNumber);
+        $this->getElement('phone_number')->setValue($phoneNumber);
     }
 
     public function verifyPassword(?string $password): void
     {
-        $this->getDocument()->fillField('Verification', $password);
+        $this->getElement('password_verification')->setValue($password);
     }
 
     public function subscribeToTheNewsletter(): void
     {
-        $this->getDocument()->checkField('Subscribe to the newsletter');
+        $this->getElement('newsletter')->check();
     }
 
     protected function getDefinedElements(): array
     {
         return array_merge(parent::getDefinedElements(), [
-            'email' => '#sylius_customer_registration_email',
-            'first_name' => '#sylius_customer_registration_firstName',
-            'last_name' => '#sylius_customer_registration_lastName',
-            'password' => '#sylius_customer_registration_user_plainPassword_first',
-            'password_verification' => '#sylius_customer_registration_user_plainPassword_second',
-            'phone_number' => '#sylius_customer_registration_phoneNumber',
+            'email' => '[data-test-email]',
```

Comment:
Maybe we can process some helper here? It could be done in the next iteration.

Reply:
Something like this:
```suggestion
            'email' => TestAttribute::string('email'),
```

Reply:
I get the idea here but I'd leave it for later when we will have more experience with the new system so that we create right abstraction layer.